precode.go

```diff
@@ -55,7 +77,20 @@ func main() {
 	var wg sync.WaitGroup
 
 	// 4. Собираем числа из каналов outs
-	// ...
+	for pos, number := range outs {
+		wg.Add(1)
+		go func(in <-chan int64, out chan<- int64, ct int) {
+			defer wg.Done()
+			for {
+				num, ok := <-in
+				if !ok {
+					return
+				}
+				amounts[ct]++
+				out <- num
```

Comment:
Можно просто сделать сразу выходной канал, без передачи его как аргумент
```suggestion
				chOut <- num
```